cicd/1-setup/cicd-dependencies.template.yml

```diff
@@ -85,6 +85,35 @@ Resources:
                   - "secretsmanager:GetSecretValue"
                 Resource:
                   - !Sub arn:aws:secretsmanager:${AWS::Region}:${AWS::AccountId}:secret:cicd/*
+  
+  # This role will be used by CodeBuild to verify pull requests. As such it has
+  # very limited permissions, as any GitHub user can create a Pull Requests.
+  # Since we may want to create multiple codebuild projects for different
+  # branches, we can append policies when those stack resources are created
+  # (in "2-cicd/cicd.template.yml")
+  PublicCodeBuildRole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Statement:
+          - Action: ['sts:AssumeRole']
+            Effect: Allow
+            Principal:
+              Service:
+                - codebuild.amazonaws.com
+        Version: '2012-10-17'
+      Path: /service-role/
+      PermissionsBoundary: !ImportValue IAM-DevPermissions
+      Policies:
+        - PolicyName: PublicCodeBuildSecretsAccess
+          PolicyDocument:
+            Statement:
+              - Effect: Allow
+                Action:
+                  - "secretsmanager:GetSecretValue"
+                Resource:
+                  - !Sub arn:aws:secretsmanager:${AWS::Region}:${AWS::AccountId}:secret:cicd/docker-hub/username-ITBwTx
+                  - !Sub arn:aws:secretsmanager:${AWS::Region}:${AWS::AccountId}:secret:cicd/docker-hub/read-only-token-SsbSj7
```

Comment:
Remind me again was the reason for the random letters at the end?  Was it the version of the secret?

Should we do this to future proof?
```
                  - !Sub arn:aws:secretsmanager:${AWS::Region}:${AWS::AccountId}:secret:cicd/docker-hub/read-only-token*
```

Reply:
>Secrets Manager includes six random characters at the end of the secret name to help ensure that the secret ARN is unique. If the original secret is deleted, and then a new secret is created with the same name, the two secrets have different ARNs because of these characters. Users with access to the old secret don't automatically get access to the new secret because the ARNs are different.

https://docs.aws.amazon.com/secretsmanager/latest/userguide/getting-started.html

I'm of two minds on this. We could future proof it, or we could prefer that it break :man_shrugging:

Reply:
Yeah, I'd say leave it as is.  Thanks!